lib/acpi/acpi.c

```diff
@@ -470,7 +469,7 @@ int acpi_device_irq_get(struct acpi_dev *child_dev, struct acpi_irq_resource *ir
 		irq_res->flags = arch_acpi_encode_irq_flags(res->Data.ExtendedIrq.Polarity,
 							    res->Data.ExtendedIrq.Triggering);
 	} else {
-		if (res->Data.Irq.InterruptCount > CONFIG_ACPI_IRQ_VECTOR_MAX) {
+		if (res->Data.Irq.InterruptCount > irq_res->irq_vector_max) {
 			return -ENOMEM;
 		}
```

Comment:
This is fine, but you're also missing the update of `irq_res->irq_vector_max` before returning from the function (to indicate how many entries were actually filled in). Maybe the variable should be `irq_vector_count`, since it's only a maximum when calling the function, but the actual count when returning.